src/stdio/src/helpers.rs

```diff
@@ -0,0 +1,159 @@
+use super::{internal, BUFSIZ, FILE, UNGET};
+use compiler_builtins::mem::memset;
+use stdlib::calloc;
+use core::{mem, ptr};
+use core::sync::atomic::AtomicBool;
+use platform::types::*;
+use super::constants::*;
+use fcntl::*;
+use platform;
+use errno;
+
+/// Parse mode flags as a string and output a mode flags integer
+pub unsafe fn parse_mode_flags(mode_str: *const c_char) -> i32 {
+    use string::strchr;
+    let mut flags = if !strchr(mode_str, b'+' as i32).is_null() {
```

Comment:
The man page for `fopen` states

> The argument mode points to a string beginning with one of the  following  sequences
> r ...
> r+ ...
> w ...
> w+ ...
> a ...
> a+ ...

We don't make the same guarantee since we use `strchr`. Is that intentional?

Reply:
refer to `_fdopen()` and `fopen`. Both of them guarentee that either 'r', 'w' or 'a' are at the start of the string. You must also consider things like `rb+`, so hard coding the '+' to be `mode_str[1]` would not be a good idea for future implementation